Micro batch merger controller docker

bashir2 opened the pull request:
<!-- This is based on openmrs-cord PR template. -->

## Description of what I changed
Fixes #360 

This adds a `Dockerfile` for the controller and a `docker-compose` config for combining this new docker
image with Spark images (from Bitnami). It also adds an option for limiting number of pipeline workers.

<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->
<!--- If your PR is related to an issue , please mention it here. -->
<!--- It is generally a good practice to first file an issue with enough
  context and reference it in the PR, but if you don't have that, please remove
  this section. -->

## E2E test

<!-- There are different scenarios for using the tools in this repo; please
  help your reviewers by describing how you have e2e tested your change. -->

TESTED:

Ran:
```
docker-compose -f compose-controller-spark-sql.yaml up
```
Then ran the pipeline through the controller UI and query it using `beeline`
```
docker exec -it spark-thriftserver bin/beeline
```
## Checklist: I completed these to help reviewers :)

<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->

- [x] I have read and will follow the [review process](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/doc/review_process.md).
- [x] I am familiar with Google Style Guides for the language I have coded in.
- [x] My IDE is configured to follow the Google
- [ ] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and
      added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

omarismail94 commented on pipelines/batch/Dockerfile:
```diff
@@ -17,7 +17,7 @@
 
 
-FROM adoptopenjdk/maven-openjdk11
+FROM eclipse-temurin:11-jdk-focal
```
can we change to this in all the other Dockerfiles in the repo?

bashir2 replied:
This was actually causing a weird problem ([example](https://pantheon.corp.google.com/cloud-build/builds;region=global/c47bb87d-a9fa-4be1-a22e-81d0347f7796;step=17?project=cloud-build-fhir)), i.e., a DB connection issue that I had no explanation for. With trial and error I reverted some of the changes in this PR and finally reverting this, fixed the problem. For now I replace this with a TODO.

omarismail94 commented on pipelines/controller/Dockerfile:
```diff
@@ -0,0 +1,25 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM eclipse-temurin:11-jdk-focal
```
add comment on what this dockerfile does

omarismail94 reviewed (commented):
I couldn't make a comment in the file, but what is the reason behind changing the filename to `docker/DEPRECATED-docker-compose.yaml`

bashir2 reviewed (commented):
Thanks for the review; PTAL.
The reason for `DEPRECATED-docker-compose.yaml` rename is that we are actually recommending that approach anymore. So we should either remove it, or clean-up/update our streaming approach.

bashir2 commented:
To be clear, [this line](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/pull/413/commits/e613c544e3ddab51f14c884891f07ad9a8df8b1c#diff-beefc20bb55eda6b6af8f2bd6378c297febfeb10f94334948a56fe790abd2d1aL20) is the only real change since last review round.

omarismail94 reviewed (approved):
Thanks Bashir!
